nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-documentation/src/main/java/org/apache/nifi/documentation/html/AbstractHtmlDocumentationWriter.java

```diff
@@ -0,0 +1,356 @@
+        xmlStreamWriter.writeEndElement();
+    }
+
+    /**
+     * Gets the class name of the component.
+     *
+     * @param component the component to describe
+     * @return the class name of the component
+     */
+    abstract String getTitle(final T component);
+
+    /**
+     * Writes the body section of the documentation, this consists of the component description, the tags, and the PropertyDescriptors.
+     *
+     * @param component            the component to describe
+     * @param xmlStreamWriter      the stream writer
+     * @param hasAdditionalDetails whether there are additional details present or not
+     * @throws XMLStreamException thrown if there was a problem writing to the XML stream
+     */
+    void writeBody(final T component, final XMLStreamWriter xmlStreamWriter, final boolean hasAdditionalDetails) throws XMLStreamException {
+        xmlStreamWriter.writeStartElement("body");
+        writeHeader(component, xmlStreamWriter);
+        writeDeprecationWarning(component, xmlStreamWriter);
+        writeDescription(component, xmlStreamWriter, hasAdditionalDetails);
+        writeTags(component, xmlStreamWriter);
+        writeProperties(component, xmlStreamWriter);
+        writeDynamicProperties(component, xmlStreamWriter);
+        writeAdditionalBodyInfo(component, xmlStreamWriter);
+        writeStatefulInfo(component, xmlStreamWriter);
+        writeRestrictedInfo(component, xmlStreamWriter);
+        writeInputRequirementInfo(component, xmlStreamWriter);
+        writeUseCases(component, xmlStreamWriter);
+        writeMultiComponentUseCases(component, xmlStreamWriter);
+        writeSystemResourceConsiderationInfo(component, xmlStreamWriter);
+        writeSeeAlso(component, xmlStreamWriter);
+        xmlStreamWriter.writeEndElement();
+    }
+
+    /**
+     * Write the header to be displayed when loaded outside an iframe.
+     *
+     * @param component       the component to describe
+     * @param xmlStreamWriter the stream writer to use
+     * @throws XMLStreamException thrown if there was a problem writing the XML
+     */
+    private void writeHeader(final T component, XMLStreamWriter xmlStreamWriter) throws XMLStreamException {
+        xmlStreamWriter.writeStartElement("h1");
+        xmlStreamWriter.writeAttribute(ID, "nameHeader");
+        // Style will be overwritten on load if needed
+        xmlStreamWriter.writeAttribute("style", "display: none;");
+        xmlStreamWriter.writeCharacters(getTitle(component));
+        xmlStreamWriter.writeEndElement();
+    }
+
+    /**
+     * Writes a warning about the deprecation of a component.
+     *
+     * @param component       the component to describe
+     * @param xmlStreamWriter the stream writer
+     * @throws XMLStreamException thrown if there was a problem writing to the XML stream
+     */
+    abstract void writeDeprecationWarning(final T component, final XMLStreamWriter xmlStreamWriter) throws XMLStreamException;
```

Comment:
It would be helpful to group all `abstract` methods together, after all of the concrete methods.